react/features/local-recording/recording/flac/FlacAdapter.js

```diff
@@ -0,0 +1,170 @@
+            // only when flac recording is in use.
+            try {
+                // try load the minified version first
+                this._encoder = new Worker('/libs/flacEncodeWorker.min.js');
+            } catch (exception1) {
+                // if failed, try un minified version
+                try {
+                    this._encoder = new Worker('/libs/flacEncodeWorker.js');
+                } catch (exception2) {
+                    logger.error('Failed to load flacEncodeWorker.');
+                    reject();
+                }
+            }
+
+            // set up listen for messages from the WebWorker
+            this._encoder.onmessage = e => {
+                if (e.data.command === WORKER_BLOB_READY) {
+                    // receiving blob
+                    this._data = e.data.buf;
+                    if (this._stopPromiseResolver !== null) {
+                        this._stopPromiseResolver();
+                        this._stopPromiseResolver = null;
+                    }
+                } else if (e.data.command === DEBUG) {
+                    logger.log(e.data);
+                } else if (e.data.command === WORKER_LIBFLAC_READY) {
+                    logger.debug('libflac is ready.');
+                    resolve();
+                } else {
+                    logger.error(
+                        `Unknown event
+                        from encoder (WebWorker): "${e.data.command}"!`);
+                }
+            };
+
+            this._encoder.postMessage({
+                command: MAIN_THREAD_INIT,
+                config: {
+                    sampleRate: 44100,
+                    bps: 16
+                }
+            });
+        });
+
+        const callbackInitAudioContext = (resolve, reject) => {
+            navigator.getUserMedia(
+
+                // constraints - only audio needed for this app
+                {
+                    audioBitsPerSecond: 44100, // 44 kbps
+                    audio: true,
+                    mimeType: 'application/ogg' // useless?
+                },
+
+                // Success callback
+                stream => {
+                    this._audioContext = new AudioContext();
+                    this._audioSource
+                     = this._audioContext.createMediaStreamSource(stream);
+                    this._audioProcessingNode
+                      = this._audioContext.createScriptProcessor(4096, 1, 1);
+                    this._audioProcessingNode.onaudioprocess = e => {
+                        // delegate to the WebWorker to do the encoding
+                        const channelLeft = e.inputBuffer.getChannelData(0);
+
+                        this._encoder.postMessage({
+                            command: MAIN_THREAD_NEW_DATA_ARRIVED,
+                            buf: channelLeft
+                        });
+                    };
+                    logger.debug('AudioContext is set up.');
+                    resolve();
```

Comment:
Considering the density of initialization here, my advice is pulling out the logic into a named helper might be useful to contextualize a chunk of code while hiding details; assuming it creates a "cleaner" reading approach.